### PR TITLE
feat: bump up notation-go to latest version

### PR DESCRIPTION
**Pull request by binbin-li**
### What?
Fix the build failure due to the breaking change in `notation-go` as https://github.com/notaryproject/notation/issues/247 mentioned.

Related changes in `notation-go`:
https://github.com/notaryproject/notation-go/pull/77
https://github.com/notaryproject/notation-go/pull/67
https://github.com/notaryproject/notation-go/pull/64

### Test
1. Run `go clean -modcache && go mod tidy`
2. Build cmd with no errors

Signed-off-by: Binbin Li <libinbin@microsoft.com>

### CODE REVIEW

File: cmd/notation/pull.go [hunk 1]
```diff
@@ -53,13 +53,14 @@ func runPull(ctx *cli.Context) error {
 		return err
 	}
 
-	sigDigests, err := sigRepo.Lookup(ctx.Context, manifestDesc.Digest)
+	sigManifests, err := sigRepo.ListSignatureManifests(ctx.Context, manifestDesc.Digest)
 	if err != nil {
-		return fmt.Errorf("lookup signature failure: %v", err)
+		return fmt.Errorf("list signature failure: %v", err)
```
Comment (priteshbandi) [inline cmd/notation/pull.go:58]:
```suggestion
		return fmt.Errorf("list signature manifests failure: %v", err)
```

File: cmd/notation/push.go [hunk 2]
```diff
@@ -83,14 +80,11 @@ func pushSignature(ctx *cli.Context, ref string, sig []byte) (notation.Descripto
...
-	sigDesc, err := sigRepo.Put(ctx.Context, sig)
-	if err != nil {
-		return notation.Descriptor{}, fmt.Errorf("push signature failure: %v", err)
-	}
-	desc, err := sigRepo.Link(ctx.Context, manifestDesc, sigDesc)
+	// pass in nonempty annotations if needed
+	sigDesc, _, err := sigRepo.PutSignatureManifest(ctx.Context, sig, manifestDesc, make(map[string]string))
 	if err != nil {
-		return notation.Descriptor{}, fmt.Errorf("link signature failure: %v", err)
+		return notation.Descriptor{}, fmt.Errorf("put signature failure: %v", err)
```
Comment (priteshbandi) [inline cmd/notation/push.go:86]:
```suggestion
		return notation.Descriptor{}, fmt.Errorf("put signature manifest failure: %v", err)
```

File: cmd/notation/pull.go (hunk 1 above)
Comment (binbin-li) [reply cmd/notation/pull.go:58]: thanks for catching!

**Review by priteshbandi — APPROVED** [review]
LGTM

Once this PR is merged, we can close https://github.com/notaryproject/notation/pull/234

**Review by gokarnm — APPROVED** [review]
LGTM!

**Pull request merged**